clients/rospy/src/rospy/timer.py

```diff
@@ -204,7 +207,14 @@ def shutdown(self):
         Stop firing callbacks.
         """
         self._shutdown = True
-        
+        self._period = self.__period
+
+    def startup(self):
```

Comment:
It is not clear to me how this is supposed to be used by the user. If the timer has been shut down before and the thread has terminated already calling this function has no effect on its own. The user also has to call `start()` again.

If the thread is still running calling this method *undos* a previous `shutdown` call which seems to be unintuitive.